src/Router/Router.php

```diff
@@ -113,6 +114,10 @@ public function findRoute()
 
         $currentRoute = $this->currentRoute();
 
+	    if (is_array($currentRoute) && key_exists('shouldCache', $currentRoute)){
```

Comment:
This section should be placed after this condition, and there wouldn't need to check it's array or not

```
if (!$currentRoute) {
    throw RouteException::incorrectMethod($this->request->getMethod());
}
```